src/Arcus.BackgroundJobs.Tests.Integration/AzureActiveDirectory/ClientSecretExpirationJobTests.cs

```diff
@@ -39,49 +39,65 @@ public ClientSecretExpirationJobTests(ITestOutputHelper outputWriter)
-                .Build();
-
+            int expirationThreshold = 14;
+            IEventGridPublisher eventGridPublisher = CreateEventGridPublisher();
             var options = new WorkerOptions();
             options.ConfigureLogging(_logger)
                    .AddSingleton<IEventGridPublisher>(eventGridPublisher)
-                   .AddClientSecretExpirationJob(
-                        options => {
-                            options.RunImmediately = true;
-                            options.ExpirationThreshold = 14;
-                            options.EventUri = new Uri("https://github.com/arcus-azure/arcus.backgroundjobs");
-                        });
+                   .AddClientSecretExpirationJob(opt => 
+                   {
+                       opt.RunImmediately = true;
+                       opt.ExpirationThreshold = expirationThreshold;
+                       opt.EventUri = new Uri("https://github.com/arcus-azure/arcus.backgroundjobs");
+                   });
+
+            var tenantId = _config.GetValue<string>("Arcus:AzureActiveDirectory:TenantId");
+            var clientId = _config.GetValue<string>("Arcus:AzureActiveDirectory:ServicePrincipal:ClientId");
+            var clientSecret = _config.GetValue<string>("Arcus:AzureActiveDirectory:ServicePrincipal:ClientSecret");
 
             // Act
+            using (TemporaryEnvironmentVariable.Create(EnvironmentVariables.AzureTenantId, tenantId))
+            using (TemporaryEnvironmentVariable.Create(EnvironmentVariables.AzureServicePrincipalClientId, clientId))
+            using (TemporaryEnvironmentVariable.Create(EnvironmentVariables.AzureServicePrincipalClientSecret, clientSecret))
             await using (var worker = await Worker.StartNewAsync(options))
             {
                 await using (var consumer = await TestServiceBusEventConsumer.StartNewAsync(_config, _logger))
                 {
                     // Assert
                     EventBatch<Event> eventBatch = consumer.Consume();
-                    Event @event = Assert.Single(eventBatch.Events);
-                    CloudEvent cloudEvent = @event.AsCloudEvent();
-                    Assert.NotNull(cloudEvent.Id);
-                    Assert.True(cloudEvent.Type == ClientSecretExpirationEventType.ClientSecretAboutToExpire.ToString() || cloudEvent.Type == ClientSecretExpirationEventType.ClientSecretExpired.ToString());
-
-                    var cloudEventData = cloudEvent.GetPayload<ApplicationWithExpiredAndAboutToExpireSecrets>();
-                    Assert.IsType<Guid>(cloudEventData.KeyId);
-                    
-                    if (cloudEvent.Type == ClientSecretExpirationEventType.ClientSecretAboutToExpire.ToString())
+                    Assert.NotEmpty(eventBatch.Events);
+                    IEnumerable<CloudEvent> cloudEvents = eventBatch.Events.Select(ev => ev.AsCloudEvent());
+                    Assert.All(cloudEvents, ev =>
```

Comment:
Changed this to handle multiple events. If we add more expired secrets in the future, this is already taking that into account.